libraries/botbuilder/src/skills/skillHandler.ts

```diff
@@ -323,13 +325,11 @@ export class SkillHandler extends ChannelServiceHandler {
                 claimsIdentity.claims
             )}`;
 
-            return context.updateActivity(newActivity);
+            resourceResponse = await context.updateActivity(newActivity);
         });
 
-        // Note: the original activity ID is passed back here to provide "behavioral" parity with the C# SDK. Due to
-        // some inconsistent method signatures, the proper response is not propagated back through `context.updateActivity`
-        // so we have to manually pass this value back.
-        return { id: activityId };
+        // Due to a backwards-compat function signature, resourceResponse may not be defined.
```

Comment:
This was the primary blocker and lack of utility from the actual return value that kept us from taking this change; we were concerned about breaking people who implemented their own subclass of BotAdapter.

Apparently we just didn't devote enough time to thinking of another approach 😅 🤦‍♂️

Reply:
Just triple checked by extending from BotAdapter in the consumer-test script and making the `updateActivity` signature return just `Promise<void>`. Compilation with strict mode worked no problem. Changing the signature to return `Promise<boolean>` yields a compilation error 👍